canary-checker/docs/scripting/cel.md

```diff
@@ -1478,1840 +867,1991 @@ Parsing with hour and minute for a specific zone:
 
-Where:
+Syntax:
 
-- `duration` is the string representation of the duration.
+    filepath.Split(path) (dir, file)
 
 Examples:
 
 ```javascript
-Parsing a duration string:
-time.ParseDuration("1h30m")  // Evaluates to a duration of 1 hour and 30 minutes
-
-Another example with a different format:
-time.ParseDuration("15m30s")  // Evaluates to a duration of 15 minutes and 30 seconds
-
-Parsing a negative duration:
-time.ParseDuration("-2h45m")  // Evaluates to a duration of -2 hours and -45 minutes
+filepath.Split("/foo/bar/baz"); // Evaluates to [/foo/bar/ baz]
 ```
 
 ---
 
-### time.Since
+## JSON
 
-The `time.Since` function in CEL calculates the duration that has elapsed since a given time. It is commonly used to measure the time difference between a specified time and the current moment.
+### jq
+
+The `jq` function in CEL applies a jq expression to filter or transform data.
 
 Syntax:
 
-    time.Since(pastTime)
+    jq(jqExpr, data)
 
 Where:
 
-- `pastTime` is a `time.Time` object representing a past point in time.
+- `jqExpr` is the jq expression you're applying.
+- `data` is the data you're filtering or transforming.
 
 Examples:
 
 ```javascript
-Calculating the time elapsed since a specific past time:
-time.Since(time.Parse("2006-01-02", "2023-09-26"))  // Evaluates to the duration since September 26, 2023
+Filtering data with a jq expression:
+jq(".name", {"name": "John", "age": 30})  // Evaluates to "John"
 
-Another example with a different past time:
-time.Since(time.Parse("15:04 02-01-2006", "14:30 26-09-2023"))  // Evaluates to the duration since 14:30 on September 26, 2023
+Transforming data with a jq expression:
+jq("{name, age}", {"name": "John", "age": 30, "city": "NY"})  // Evaluates to {"name": "John", "age": 30}
 
-Using the `time.Now` function for a real-time duration:
-time.Since(time.Now())  // Always evaluates to a very small duration, as it's the time since "now"
+Using a complex jq expression:
+jq(".[] | select(.age > 25)", [{"name": "John", "age": 30}, {"name": "Jane", "age": 25}])  // Evaluates to [{"name": "John", "age": 30}]
 ```
 
----
-
-### time.Until
+### JSON
 
-The `time.Until` function in CEL calculates the duration remaining until a specified future time. It helps in determining the time left for an event or deadline.
+The `JSON` function in CEL is used to convert a JSON formatted string into a map. It returns a map containing the JSON data.
 
 Syntax:
 
-    time.Until(futureTime)
+    JSON(jsonString)
 
 Where:
 
-- `futureTime` is a `time.Time` object representing a future point in time.
+- `jsonString` is the JSON formatted string.
 
 Examples:
 
 ```javascript
-Calculating the time remaining until a specific future time:
-time.Until(time.Parse("2006-01-02", "2023-10-01"))  // Evaluates to the duration until October 1, 2023
+Converting a JSON string to a map:
+JSON("{\"name\": \"Alice\", \"age\": 30}")  // Evaluates to a map with keys "name" and "age"
```

Comment:
Can we use the piped version by default .e.g 

`"{\"name\": \"Alice\", \"age\": 30}".JSON() `

and then

use the comment to display the exact result, which negates the need for a comment most of the time, if a comment is needed before, then it should be a comment and not just text.  i.e. the expression should be copy and pasteable

Reply:
https://github.com/flanksource/gomplate/pull/37